client/marketing/coupons/index.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * External depenencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * WooCommerce dependencies
+ */
+import { getSetting } from '@woocommerce/wc-admin-settings';
```

Comment:
You should use the settings store. `allowMarketplaceSuggestions` could change and this value will be outdated. I should have some docs on this, sorry about that. I don't think it matters too much now because there are so many page refreshes, but if that ever changes, this will introduce a bug. Probably best to do in a follow up.